agent.go

```diff
@@ -0,0 +1,90 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ Agents = (*agents)(nil)
+
+// Agents describes all the agent-related methods that the
+// Terraform Cloud API supports.
+// TFE API docs: https://www.terraform.io/docs/cloud/api/agents.html
+type Agents interface {
+	// Read an agent by its ID.
+	Read(ctx context.Context, agentID string) (*Agent, error)
+
+	// List all the agents of the given pool.
+	List(ctx context.Context, agentPoolID string, options *AgentListOptions) (*AgentList, error)
+}
+
+// agents implements Agents.
+type agents struct {
+	client *Client
+}
+
+// AgentList represents a list of agents.
+type AgentList struct {
+	*Pagination
+	Items []*Agent
+}
+
+// Agent represents a Terraform Cloud agent.
+type Agent struct {
+	ID         string `jsonapi:"primary,agents"`
+	Name       string `jsonapi:"attr,name"`
+	IP         string `jsonapi:"attr,ip-address"`
+	Status     string `jsonapi:"attr,status"`
+	LastPingAt string `jsonapi:"attr,last-ping-at"`
+}
+
+type AgentListOptions struct {
+	ListOptions
+
+	//Optional:
+	Filter string `url:"filter[last-ping-since],omitempty"`
```

Comment:
This field value is supposed to be a date in ISO8601 format (ex. 2020-08-11T10:41:23Z), not just any string. It would be helpful to know what was expected. Try this encoding instead:

LastPingSince        time.Time            `url:"filter[last-ping-since],omitempty,iso8601"`

Reply:
Lol that's actually a mistake. Thanks for catching!